### DISCUSSION TITLE
Is there a way to get specifics on what went wrong with the validation?

### DISCUSSION THREAD

**Discussion by DanielCoder834** [go-playground/validator#1303] [Q&A] [answered]
For example, a struct like this: 

```
type UpdateEmail struct {
	Email    string `validate:"required, email"`
	NewEmail string `validate:"required, email"`
        Useername string `validate:"required"`
}
```
If a user entered an invalid email, is there a way to see which email field is invalid? In addition to that, is it possible to see what specifically went wrong with the validation, like if one of the fields was nil or another issue? Thank you for any help anyone can provide on this question, it means a lot. :)

**Comment by deankarn** [comment, marked as answer]
Please see the documentation and examples folder.

They will show you how to pull out the specifics of all errors that occurred during the validation including which field and validation failed for it.

**Reply by DanielCoder834** [reply]
ok thank you
